module/solutions/helper.js

```diff
@@ -694,4 +743,280 @@ module.exports = class SolutionsHelper {
+     * @param {Object} solutionData - new solution creation data
+     * @param {Boolean} [isAPrivateProgram = false] - created program is private or not     
+     * @returns {Object} Created solution and program
+     */
+
+    static createProgramAndSolutionFromTemplate(
+      templateId,
+      program,
+      userId,
+      solutionData,
+      isAPrivateProgram = false
+  ) {
+      return new Promise(async (resolve, reject) => {
+          try {
+
+              let dateFormat = gen.utils.epochTime();
+              let programData;
+
+              if( program.id === "" ) {
+
+                programData = await programsHelper.create({
+                  externalId : program.name + "-" + dateFormat ,
+                  description : solutionData.description,
+                  name : program.name,
+                  userId : userId,
+                  isAPrivateProgram : isAPrivateProgram
+                });
+                
+                program.id = programData._id;
+              }
+
+              let solutionQuery =   {
+                _id : templateId
+              };
+
+              let programQuery = {
+                _id : program.id
+              };
+
+              let duplicateSolution = 
+              await this.importFromSolution(
+                solutionQuery,
+                programQuery,
+                userId,
+                solutionData,
+                false
+              );
+
+              return resolve(
+                _.pick(
+                  duplicateSolution,
+                  [
+                    "_id",
+                    "externalId",
+                    "frameworkExternalId",
+                    "frameworkId",
+                    "programExternalId",
+                    "programId",
+                    "entityTypeId",
+                    "entityType",
+                    "isAPrivateProgram"
+                  ]
+                  ));
+
+          } catch (error) {
+              return reject(error);
+          }
+      })
+  }
+
+      /**
+     * Import solution from template.
+     * @method
+     * @name importFromSolution -
+     * @param {Object} solutionQuery - solution query data
+     * @param {Object} programQuery - program query data.
+     * @param {String} userId - logged in user id.
+     * @param {Object} data - new solution data  
+     * @returns {Object} New solution information
+     */
+
+    static importFromSolution(
+      solutionQuery,
+      programQuery,
+      userId,
+      data,
+      isReusable = true
+    ) {
+      return new Promise(async (resolve, reject) => {
+        try {
+
+          let solutionDocument = await this.solutionDocuments(
+            solutionQuery
+          );
+  
+          if (!solutionDocument[0]._id) {
+            throw messageConstants.apiResponses.SOLUTION_NOT_FOUND;
+          }
+  
+          let programDocument = 
+          await programsHelper.list(
+            programQuery, 
```

Comment:
@aman-tunerlabs - Why to expect a query, can it not be just programExternalId

Reply:
@aks30  As frontend will pass _id in create api for program. So instead of making a database call to get programExternalId i made it to work for both programExternalId and program _id.

Reply:
@aman-tunerlabs - Same function is called using 2 different ids i.e. one calls with external and other with internal id ?

Reply:
@akash-above yes